### PR TITLE
Subtree copy

### PR DESCRIPTION
**Pull request by adamwojs**
# Description

This PR implements actor to perform subtree copy for SPI and PAPI. Needed for https://github.com/ezsystems/PlatformUIBundle/pull/893 to determine a limit of number of nodes allowed to copy at once. 

# TODO
- [X] Implementation
- [x] Tests
- [x] Update README

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -3,11 +3,13 @@
...
     "require": {
         "php": "~5.4 || ~7.0",
-        "ezsystems/ezpublish-kernel": "~5.3.5@dev || ~5.4.2@dev || ^6.0@dev",
+        "ezsystems/ezpublish-kernel": "~5.3.5@dev || ~5.4.2@dev || ~6.11@dev",
         "fzaninotto/faker": "~1.4",
         "jdorn/sql-formatter": "~1.2"
     },
-
+    "require-dev": {
+        "phpunit/phpunit": "^4.8.35"
```
Comment (alongosz) [inline composer.json]: Let's try to use PHPUnit 6 since we don't have existing tests (right? ;) ), 
but also let's make sure that this doesn't create conflict when required from meta-repo.

File: phpunit.xml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        <testsuite name="eZ\Publish\Profiler">
+            <directory>src/php/eZ/Publish/Profiler/Tests/</directory>
+        </testsuite>
+        <testsuite name="eZ\Publish\ProfilerBundle">
+            <directory>src/php/eZ/Publish/Core/Repository/Tests</directory>
+        </testsuite>
+    </testsuites>
+    <filter>
+        <whitelist>
+            <directory>./</directory>
```
Comment (alongosz) [inline phpunit.xml]: I'd go here for just `src` and remove `vendor` exclusion a few lines below.

**Review by alongosz — CHANGES_REQUESTED** [review]
Review on tests (6c663b1): they look great
I have some doubts about config (as discussed, but posting them here so we don't miss / forget anything):

File: composer.json [hunk 3]
```diff
@@ -3,11 +3,13 @@
     "type": "ezplatform-bundle",
     "require": {
         "php": "~5.4 || ~7.0",
-        "ezsystems/ezpublish-kernel": "~5.3.5@dev || ~5.4.2@dev || ^6.0@dev",
+        "ezsystems/ezpublish-kernel": "~5.3.5@dev || ~5.4.2@dev || ~6.0@dev",
```
Comment (andrerom) [inline composer.json]: why this change?

File: src/php/eZ/Publish/Profiler/Actor/Handler/Exception/ActorHandlerException.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+use Throwable;
+
+class ActorHandlerException extends RuntimeException
+{
+    /**
+     * @var \eZ\Publish\Profiler\Actor
+     */
+    protected $actor;
+
+    public function __construct(Actor $actor, $message = '', $code = 0, Throwable $previous = null)
```
Comment (andrerom) [inline src/php/eZ/Publish/Profiler/Actor/Handler/Exception/ActorHandlerException.php]: php7 code detected

**Review by andrerom — CHANGES_REQUESTED** [review]
Some questions/comments.

Also on the testing, considered phpspec as alternative here given there is nothing from before? See hybrid-ui for example, might be easier for the test usage here.

File: composer.json (hunk 3 above)
Comment (adamwojs) [reply composer.json]: Ups, this should not be committed ;-)

**Comment by adamwojs** [comment]
@andrerom I think that most of the tests here will be about creating mocks, so use phpspec here might be a good idea...

**Comment by andrerom** [comment]
> I think that most of the tests here will be about creating mocks, so use phpspec here might be a good idea...

I guess you meant to say `_not_ a good idea` here right? Fine with me, just wanted to see if we could avoid having test with intimate knowledge of the internal _(as it is non-optimal when you later needing to refactor or fix something)_, hence use spec.

**Comment by adamwojs** [comment]
Close as obsolete
